packages/devtools_app/lib/src/screens/vm_developer/object_inspector/class_hierarchy_explorer.dart

```diff
@@ -35,7 +37,9 @@ class ClassHierarchyExplorer extends StatelessWidget {
           controller.classHierarchyController.selectedIsolateClassHierarchy,
       dataDisplayProvider: (node, onPressed) => VmServiceObjectLink<Class>(
         object: node.cls,
-        onTap: controller.findAndSelectNodeForObject,
+        onTap: (cls) => unawaited(
+          controller.findAndSelectNodeForObject(context, cls),
```

Comment:
passing BuildContext around is dangerous. As this context could become disposed while the controller is still trying to use it. Can we access the router without BuildContext? Or if that is not possible, can we lookup the router in widget code so that we don't have a potentially stale build context floating around?

Reply:
Unfortunately, `Router.navigate` is a static method that requires a `BuildContext`, so there's no way to even look up the router and cache it for later :(

Reply:
Then if possible, we should have the widget code handle the calls to Router.navigate. We don't want to be passing around BuildContext outside of widget code. The widgets could respond to an event from the controller, and then call the `Router.navigate(context ...)` from somewhere where the build context is accessible (e.g. a StatefulWidget)